controllers/managedocs_controller.go

```diff
@@ -311,10 +317,18 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager, ctrlOptions *c
 			&source.Kind{Type: &opv1a1.ClusterServiceVersion{}},
 			enqueueManangedOCSRequest,
 			csvPredicates,
-		).
+		)
+
+	r.Log.Info(fmt.Sprintf("Creating ManagedOCS %s controller", r.NetworkType))
+
+	if r.NetworkType == OVNNetworkType {
+		controller = controller.Owns(&ovnv1.EgressFirewall{})
+	} else {
+		controller = controller.Owns(&openshiftv1.EgressNetworkPolicy{})
+	}
```

Comment:
If we are to use an enum, please use a switch statement that covers all option and a default option if the value is not supported that return an error (or even Panic)